config/set_config.py

```diff
@@ -0,0 +1,148 @@
+"""
+A script for inserting frontend config files via the `config/frontend` PUT endpoint.
+
+# Example
+
+```sh
+python3 config/set_config.py --retry dev
+```
+
+Will try and set the config for the dev environment
+by reading all the settings and the configs from the `<repo root>/config/dev/` directory.
+
+The tool will find the endpoint, and any other config it needs to operate from the `settings.json` file.
+It will apply `config.json` as the default config, and if any `ip_xxx.xxx.xxx.xxx.config.json` files exist,
+will also set the IP address specific configs as defined by their file name.
+"""
+
+import argparse
+import json
+import os
+import glob
+import time
+import requests
+import ipaddress
+
+
+# helpers
+def load_json_file(filepath: str):
+    with open(filepath, "r") as f:
+        return json.load(f)
+
+
+def parse_ip_config_file(path: str) -> tuple[str, dict]:
+    filename = os.path.basename(path)
+    if not filename.startswith("ip_") or not filename.endswith(".config.json"):
+        raise ValueError(f"Invalid IP config filename: {filename}")
+
+    ip_part = filename[len("ip_") : -len(".config.json")]
+
+    try:
+        ipaddress.ip_address(ip_part)
+    except ValueError:
+        raise ValueError(f"Invalid IP address in filename: {ip_part}")
+
+    return ip_part, load_json_file(path)
+
+
+def extract_attribute(obj, field_name: str):
```

Comment:
Whats the difference between this function and just using `obj[field_name]` directly in the code.  because it too will raise a key error if the key isn't present?

Remove this function, its redundant.